x/mongo/driver/topology/pool_test.go

```diff
@@ -233,13 +233,18 @@ func TestPool(t *testing.T) {
 				}
 			}()
 			go func() {
-				_, err := p.get(getCtx)
+				defer close(disconnectDone)
+				for getCtx.Err() == nil {
```

Comment:
The original problem was two things:
1) sometimes the connection gets refused for this p.get() call
2) disconnectDone wasn't getting closed, which made it wait for the test timeout when p.get() errored.

This may end up still being brittle depending on how often the connection is refused, but it shouldn't take 30 minutes for the test to error anymore.

Reply:
Why does the connection get refused in the first place?

Reply:
Took another look, closing the connection in the looping function should make sure there are always enough connections free

Reply:
It's still not clear to me what the issue was. Based on your last comment, it seems like the first goroutine would consume all three connections, which would cause the `get` call in the second goroutine to block forever. Is that correct?

Reply:
The first goroutine would consume all the connections, the `get` in the second goroutine would fail, and because `close(disconnectDone)` wasn't deferred, `disconnectDone` wouldn't get closed when the second goroutine errored, thus the first routine would keep looping forever.